app/assets/src/components/ui/controls/SampleTypeSearchBox.jsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { groupBy } from "lodash/fp";
+
+import PropTypes from "~/components/utils/propTypes";
+
+import LiveSearchPopBox from "./LiveSearchPopBox";
+
+class SampleTypeSearchBox extends React.Component {
+  handleSearchTriggered = query => {
+    const matchSampleTypes = sampleType => {
+      // Match chars in any position. Good for acronyms.
+      const regex = new RegExp(query.split("").join(".*"), "gi");
```

Comment:
are we assuming single word? What happens with spaces and word order?

Reply:
Currently, spaces are considered just another character, so the query cannot contain more spaces than in the name. For example: "ab domen" will not match. "whole b" will match. 

Having thought about it now, I think ignoring spaces in the query is more intuitive. What do you think?

Reply:
Sorry, did not quite get the example: what does "whole b" match?

Reply:
I see, "whole blood". 
I guess its more of a UX decision. IMO, I would probably split it on spaces and run the query independently for each word. Would love that "blood whole" would match too... (maybe there are other side effects im not seeing right now).

Reply:
I changed it to ignore the spaces. As for matching words in any order, it seems like overkill to me, but happy to be convinced otherwise.

Reply:
For this particular case, it probably is... for a more general function I would do it. So, up to you.